src/v/cluster/service.h

```diff
@@ -158,6 +162,7 @@ class service : public controller_service {
     ss::future<partition_state_reply>
       do_get_partition_state(partition_state_request);
 
+    std::unique_ptr<controller>& _controller;
```

Comment:
Can this be a raw pointer to the controller instead of a reference to a unique_ptr.

This class should not be able to do things like reset the pointer.